tests/framework/clients/bucket.go

```diff
@@ -71,6 +73,30 @@ func (b *BucketOperation) CheckOBC(obcName, check string) bool {
 	}
 	logger.Infof("%s resources %v all %s", obcName, resources, check)
 
+	if shouldBeBound {
+		// OBC should be in bound phase as well as existing
+		state, _ := b.k8sh.GetResource("obc", obcName, "--output", "jsonpath={.status.phase}")
+		if state != "Bound" {
+			logger.Infof("resources exist, but OBC is not in \"Bound\" phase: %s", state)
+			return false
+		}
+
+		// Regression test: OBC should have spec.objectBucketName set
+		obName, _ := b.k8sh.GetResource("obc", obcName, "--output", "jsonpath={.spec.objectBucketName}")
+		if obName == "" {
+			logger.Errorf("failed regression: OBC spec.objectBucketName is not set")
+			return false
+		}
+		// Regression test: OB should have claim ref to OBC
+		refName, _ := b.k8sh.GetResource("ob", obName, "--output", "jsonpath={.spec.claimRef.name}")
+		if refName != obcName {
+			logger.Errorf("failed regression: OB spec.claimRef.name (%q) does not match expected OBC name (%q)", refName, obcName)
+			return false
+		}
+
+		logger.Infof("OBC is \"Bound\"")
```

Comment:
nit: this is more readable IMO
```suggestion
		logger.Infof(`OBC is "Bound"`)
```